File: CalebCurry/AndroidJavaAllInOne/ActivityPractice/app/src/main/java/com/calebcurry/activitypractice/MainActivity.java
package com.calebcurry.activitypractice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
    public void postMsg(View v){
        Intent i=new Intent(this,PostMsgActivity.class);
        String s=((Button) v).getText().toString();
        i.putExtra("Name",s);
        startActivity(i);

    }
}

File: PhilippLackner(Kotlin)/episode30/app/src/main/java/com/philipplackner/episode30/MainActivity.kt
package com.philipplackner.episode30

import android.content.Context
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        /*btnStartService.setOnClickListener {
            Intent(this,MyIntentService::class.java).also {
                startService(it)
                tvServiceInfo.text="Service is running"
            }
        }
        btnStopService.setOnClickListener {
            MyIntentService.stopService()
            tvServiceInfo.text="Service is stopped"
        }*/
        btnStartService.setOnClickListener {
            Intent(this,MyService::class.java).also {
                startService(it)
                tvServiceInfo.text="Service is running"
            }
        }
        btnStopService.setOnClickListener {
            Intent(this,MyService::class.java).also {
                stopService(it)
                tvServiceInfo.text="Service is stopped"
            }
        }
        btnSendData.setOnClickListener {
            Intent(this,MyService::class.java).also {
                val dataString=etData.text.toString()
                it.putExtra("EXTRA_DATA",dataString)
                startService(it)
            }
        }
    }
}
